src/Session.hs

```diff
@@ -3,11 +3,12 @@
 -- | A persistent version of the Ghci session, encoding lots of semantics on top.
 --   Not suitable for calling multithreaded.
 module Session(
-    Session, withSession,
+    Session(allowEval), withSession,
```

Comment:
I find the approach of exposing one accessor to Session a bit unpleasant, partly because you can then mutate it later as you wish. Maybe I'm fussing though - it seems to work out relatively nicely below...

Reply:
I think my preference would be an `enableEval :: Session -> Session` though.

Reply:
Yeah, I wasn't in love with it either, but the thought of threading `allow_eval` through every function seemed less pleasant. I'm happy to put in the work if you'd prefer it -- maybe as a `ReaderT` so anyone else who might want to do so in the future will have less work?

Reply:
I was thinking of keeping allowEval in Session, but not exposing the allowEval (which gives you a lot of power), and instead just `enableEval s = s{allowEval=True}`. That way `Session` remains entirely abstract, but you can keep the rest of the code the same, without any ReaderT or similar.